### PR TITLE
Transaction multichain fix starttracking

### PR DESCRIPTION
**Pull request by shanejonas**
## Explanation
This fixes `startTrackingByNetworkClientId`'s trackingMap as well as adds `stopTrackingByNetworkClientId` and some initial tests for them both

### CODE REVIEW

File: packages/transaction-controller/src/TransactionController.test.ts [hunk 1]
```diff
@@ -4399,4 +4399,29 @@ describe('TransactionController', () => {
       Current tx status: ${TransactionStatus.submitted}`);
     });
   });
+  describe.only('startTrackinbByNetworkClientId', () => {
```
Comment (shanejonas) [inline packages/transaction-controller/src/TransactionController.test.ts:4402]:
```suggestion
  describe('startTrackinbByNetworkClientId', () => {
```

File: packages/transaction-controller/src/TransactionController.test.ts [hunk 2]
```diff
@@ -4399,7 +4399,7 @@ describe('TransactionController', () => {
       Current tx status: ${TransactionStatus.submitted}`);
     });
   });
-  describe.only('startTrackinbByNetworkClientId', () => {
+  describe('startTrackinbByNetworkClientId', () => {
```
Comment (jiexi) [inline packages/transaction-controller/src/TransactionController.test.ts:4402]:
```suggestion
  describe('startTrackingByNetworkClientId', () => {
```

File: packages/transaction-controller/src/TransactionController.ts [hunk 3]
```diff
@@ -1176,11 +1191,12 @@ export class TransactionController extends BaseControllerV1<
     this.addPendingTransactionTrackerListeners(pendingTransactionTracker);
 
     // add to tracking map
-    this.trackingMap.get(networkClientId)?.add({
+    this.trackingMap.set(networkClientId, {
       nonceTracker,
       incomingTransactionHelper,
       pendingTransactionTracker,
     });
+    return this.trackingMap;
```
Comment (jiexi) [inline packages/transaction-controller/src/TransactionController.ts:1199]: should we leave a comment that this is temporary?

File: packages/transaction-controller/src/TransactionController.ts [hunk 4]
```diff
@@ -2693,6 +2718,15 @@ export class TransactionController extends BaseControllerV1<
...
   }
 
+  private removePendingTransactionTrackerListeners(
+    pendingTransactionTracker = this.pendingTransactionTracker,
+  ) {
+    pendingTransactionTracker.hub.removeAllListeners('transaction-confirmed');
+    pendingTransactionTracker.hub.removeAllListeners('transaction-dropped');
+    pendingTransactionTracker.hub.removeAllListeners('transaction-failed');
+    pendingTransactionTracker.hub.removeAllListeners('transaction-updated');
+  }
```
Comment (jiexi) [inline packages/transaction-controller/src/TransactionController.ts:2728]: does incomingTransactionHelper.hub need to have it's listeners removed too?

**Pull request merged**